assets/js/frontend/add-to-cart.js

```diff
@@ -5,6 +5,18 @@ jQuery( function( $ ) {
 		return false;
 	}
 
+	$.wc_clear_prepend_and_scroll_to_notices = function( notices, elementAfterNotices, scrollToElementSelector ) {
```

Comment:
Does this function do too much?

Outputting the notices and scrolling to them could be a single method.

Clearing existing notices could be a separate method.

`wc_display_notices`/`wc_remove_notices`